shell/platform/android/io/flutter/plugin/platform/PlatformViewsController.java

```diff
@@ -623,30 +627,25 @@ private void initializePlatformViewIfNeeded(int viewId) {
 
     PlatformView platformView = factory.create(context, viewId, createParams);
     View view = platformView.getView();
-
-    // Reverse scale the view based on the screen density.
-    //
-    // Flow pixels are based on the physical resolution. For example, 1000 pixels in flow equals
-    // 500 points in Android if the display density is 2.0.
-    float density = getDisplayDensity();
-    view.setPivotX(0);
-    view.setPivotY(0);
-    view.setScaleX(1 / density);
-    view.setScaleY(1 / density);
-
     platformViews.put(viewId, view);
-    ((FlutterView) flutterView).addView(view);
+
+    FlutterMutatorView mutatorView = new FlutterMutatorView(context);
+    mutatorViews.put(viewId, mutatorView);
+    mutatorView.addView(platformView.getView());
+    ((FlutterView) flutterView).addView(mutatorView);
   }
 
-  public void onDisplayPlatformView(int viewId, int x, int y, int width, int height) {
+  public void onDisplayPlatformView(
+      int viewId, int x, int y, int width, int height, FlutterMutatorsStack mutatorsStack) {
     initializeRootImageViewIfNeeded();
     initializePlatformViewIfNeeded(viewId);
 
+    float density = context.getResources().getDisplayMetrics().density;
```

Comment:
Any chance we could move this to the constructor? I'm not sure how expensive is getResources or getDisplayMetrics.

Reply:
done